### PR TITLE
Tell the user why an upload was rejected

### PR DESCRIPTION
**Pull request by patrickweh**
Both upload components answered every failure with the same sentence: "Your file upload failed. Please try again." A tenant had no way to tell whether the file was too large, of the wrong type, or something else.

## Why the reason has to come from the browser

Three things were checked before writing this:

- Livewire calls the error callback with no arguments at all (`markUploadErrored` in `livewire.esm.js`), so the `error` parameter the components accepted was always `undefined`.
- The component error bag stays empty. Polled 40 times over 6 seconds in a live browser after a failed upload, it never filled.
- The 422 of the upload endpoint carries only `files.0 could not be uploaded`, which would not help anyone either.

The only place where a precise reason exists is the browser, before the upload starts, where name, size and type are known.

## What changes

`$nuxbe.validateFiles()` checks each file against the effective upload limit and the accept attribute of the input, and both components show what is actually wrong: the file name with its size and the limit, or the accepted types.

`Helper::getMaxUploadSizeInBytes()` derives that limit from the smallest of the livewire rule, `upload_max_filesize` and `post_max_size`. It parses those values itself instead of using the `fromFileSizeToBytes` macro, which throws on legal ini forms like a plain byte count, `0` or `-1`, and would have turned every page containing an upload into a 500 on such a host.

Two details worth naming: the file name is escaped, because the dialog renders its description through `x-html` and a name is fully user controlled. And the accept list is split on commas as well as semicolons, because the purchase invoice upload passes `application/pdf;image/*` and a comma only split would have rejected every file there.

## Tests

Browser tests cover size, type, the semicolon list, a file whose mime type the browser leaves empty, and the escaping. Unit tests cover the limit calculation including the ini forms that used to throw.

## Summary by Sourcery

Improve file upload UX by validating files client-side and surfacing specific rejection reasons to users.

New Features:
- Add client-side file validation for uploads based on maximum size and accepted types, returning detailed rejection metadata for the first invalid file.
- Display contextual error dialogs that name the rejected file, its size, and either the configured size limit or the allowed file types.
- Expose shared upload utilities (file-size formatter, HTML escaping, and validation) via the global $nuxbe object for reuse across upload components.

Bug Fixes:
- Prevent server errors caused by parsing edge-case PHP ini upload size values by introducing a resilient size parsing helper and using it to compute the effective upload limit.
- Ensure file names in upload error dialogs are safely escaped to avoid markup injection in HTML-rendered messages.
- Correct handling of accept lists separated by semicolons or containing wildcard and extension-based patterns so valid files are no longer incorrectly rejected.

Enhancements:
- Centralize computation of the effective maximum upload size by combining Livewire rules with PHP ini limits and using the minimum applicable value.
- Extend both media upload components to clear invalid file inputs and reuse shared client-side validation and error presentation logic.

Tests:
- Add browser tests covering client-side upload validation scenarios, including size and type rejections, semicolon-separated accept lists, empty MIME types, and HTML escaping of file names.
- Add unit tests for maximum upload size calculation and ini value parsing, including previously problematic forms like plain byte counts, zero, negative, and malformed values.

### CODE REVIEW

File: resources/js/nuxbe/uploads.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    while (size >= 1024 && unit < units.length - 1) {
+        size /= 1024;
+        unit++;
+    }
+
+    const decimals = size < 10 && unit > 0 ? 1 : 0;
+
+    const formatted = new Intl.NumberFormat(
+        document.documentElement.lang || 'de',
```
Comment (SirSplasch) [inline resources/js/nuxbe/uploads.js]: Why not fallback to 'en'?

File: resources/js/nuxbe/uploads.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+        return (
+            extensions.length === 0 ||
+            extensions.some((pattern) => name.endsWith(pattern))
+        );
+    }
+
+    return patterns.some((pattern) => {
+        if (pattern.startsWith('.')) return name.endsWith(pattern);
+        if (pattern.endsWith('/*'))
```
Comment (SirSplasch) [inline resources/js/nuxbe/uploads.js]: please add {} to the if statements and wrap it accordingly.

File: resources/js/nuxbe/uploads.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            return {
+                file,
+                reason: 'size',
+                size: formatFileSize(file.size),
+                maxSize: formatFileSize(maxSize),
+            };
+        }
+
+        if (patterns.length && !matchesAccept(file, patterns)) {
+            return { file, reason: 'type', accept: patterns.join(', ') };
```
Comment (SirSplasch) [inline resources/js/nuxbe/uploads.js]: Why is this object not wrapped, while the maxSize object is?

File: src/Helpers/Helper.php [hunk 4]
```diff
@@ -106,6 +107,27 @@ public static function classExists(string $classString,
         }
     }
 
+    public static function getMaxUploadSizeInBytes(): int
```
Comment (SirSplasch) [inline src/Helpers/Helper.php]: after getHtmlInputFieldTypes() method

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: resources/js/nuxbe/uploads.js (hunk 1 above)
Comment (SirSplasch) [reply resources/js/nuxbe/uploads.js]: Question not answered, just moved the issue to a different location.

**Review by SirSplasch — CHANGES_REQUESTED** [review]
I still need an answer on why the default locale is 'de' instead of 'en'.

**Review by SirSplasch — APPROVED** [review]
